server/game/gamesteps/sundown/discardtohandsizeprompt.js

```diff
@@ -0,0 +1,53 @@
+const BaseStep = require('../basestep');
+
+class DiscardToHandSizePrompt extends BaseStep {
+    constructor(game) {
+        super(game);
+        this.players = this.game.getPlayersInFirstPlayerOrder();
+    }
+
+    continue() {
+        while(this.players.length) {
+            let currentPlayer = this.players.shift();
+            if(! currentPlayer.isOverHandsizeLimit()) {
+                this.game.addMessage('{0} already meets their hand size limit', currentPlayer);
+            } else {
+                this.promptPlayerToDiscard(currentPlayer);
+                return false;
+            }
+        }
+        this.game.raiseEvent('onHandsizeLimitChecked');
+    }
+
+    promptPlayerToDiscard(currentPlayer) {
+        let overHandsize = currentPlayer.hand.length - currentPlayer.handSize;
+        let cardOrCards = overHandsize === 1 ? 'card' : 'cards';
+        this.game.promptForSelect(currentPlayer, {
+            ordered: true,
```

Comment:
I do not think there needs to be ordered discard at this time. I believe the only instance where it makes sense is for LeMat.

Reply:
:heavy_check_mark: